rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -144,17 +146,23 @@ def parse_changes(lines):
             (?P<what>Added|Changed|Removed)(\s+functions|variables)?
             (\s+\((?P<filtered_out>\d+)\s+filtered\s+out\))?
             ''', re.VERBOSE)
-            result = {}
+            result_list = []
+            filename = 'Undetected filename'
             for line in lines:
+                mt = title_re.match(line)
+                if mt:
+                    filename = mt.group('filename')
                 ms = summary_re.match(line)
                 if ms:
+                    result = {'filename': filename}
                     ds = ms.groupdict()
                     result[ds['kind']] = {}
                     for mc in changes_re.finditer(ds['changes']):
                         dc = mc.groupdict()
                         if int(dc['count']) or dc['filtered_out']:
                             result[ds['kind']][dc['what']] = dc
-            return result
+                    result_list.append(result)
+            return result_list
```

Comment:
Here you are returning list, but if `abipkgdiff` return code is 0, empty dict is assigned to a key in the result dictionary. That makes it inconsistent.